server/src/utils/socketIoUtils.ts

```diff
@@ -83,27 +89,24 @@ export const emitRoadmapEvent = async <T extends ClientEvents>(
 ) => {
   const {
     roadmapId,
-    dontEmitToUserId,
     requirePermission,
     event,
+    dontEmitToUserId,
+    onlyEmitToUserId,
     eventParams: parameters,
   } = { ...emitRoadmapEventParams };
 
-  console.log('Emitting roadmap socket event:');
-  console.log(JSON.stringify(emitRoadmapEventParams, undefined, 2));
-
   const roomName = `roadmap-${roadmapId}`;
   const roadmapSockets = await io.in(roomName).fetchSockets<ISocketData>();
   const socketUserIds = roadmapSockets.map((s) => s.data.user.id);
```

Comment:
```suggestion
  let socketUserIds = roadmapSockets.map((s) => s.data.user.id);
  if (dontEmitToUserId !== undefined) {
    socketUserIds = socketUserIds.filter((id) => id !== dontEmitToUserId);
  }
  if (onlyEmitToUserIds) {
    socketUsers = socketUsers.filter((user) => onlyEmitToUserIds.includes(user.id));
  }
```

It would make sense to filter out only the requested ids as soon as possible, and that would be here.

Reply:
Is there some reason why the ids could or should not be filtered here, but only after fetching from the database?

Reply:
No reason at all, a brainfart on my part. I thought I had added it before the query :sweat_smile: